### DISCUSSION TITLE
Suggestion on best way to proceed

### DISCUSSION THREAD

**Discussion by tsx-5** [ExperienceLovelace/ha-floorplan#409] [Q&A]
So,

I've got a basic floorplan for my basement working (I have 3 floors in my house).  It's ok, but looking at other examples - It could be better.  Thus, I'd like to ask for assistance again.  I've included my SVG to help with my questions...

- The first question I have is about the quality of the render, for some reason it doesn't seem as good as some of the examples I've seen online or in this group... But I can't put my finger on what is off.  I'd love thoughts on that.

- The 2nd question is regarding the how I've broken this basement out.  If you look at the SVG, you'll see that I broke it into several different "areas".  Most of these areas are not broken apart by walls.  However, I've created overlays matching up to these virtual "rooms".  That has caused the whole basement to have overlays that intersect with each other in open spaces...  It seems to me to be a bit off, but I thought I'd ask if that's the correct way to do it.  If it is not the correct way to do it...  What would be a better way to approach it?  If it is the correct way, how do I tackle opacity where the virtual boundaries meet?

Looking forward to your suggestions, and express my gratitude in advance,

![Basement-ink-floorplan-3D](https://github.com/ExperienceLovelace/ha-floorplan/assets/106412847/a9b9ffd2-86b1-4db0-96eb-aaa175a1d103)

**Comment by tsx-5** [comment]
Here's the render from SH3D

![Basement-FloorPlan3](https://github.com/ExperienceLovelace/ha-floorplan/assets/106412847/ce866d85-6b2d-49da-9db9-57747fab7ba2)

**Reply by Trzinka** [reply]
Hi!

As I already suggested, replace in CSS "fill: none !important" and do not use it, because it is not suitable in image mode (that's why such appearance no. 1 images), use "opacity" instead.

To fill the floor in SH3D, use another material "laminate, parquet or just paint (same color)".
In SH3D, it is possible to combine floors, but it is complicated, but it is possible!

Watch the video "advanced knowledge - it can confuse you" (choose the video that will help you understand what you want)
https://www.youtube.com/playlist?list=PLNaeA0iHcKdQIfb0TjdS93yqvXJHjTb45 

and 

https://github.com/Sian-Lee-SA/YouTube/tree/master/Home%20Assistant%20Floorplan
`and example (code)`

**Reply by tsx-5** [reply]
Thanks, I do like those videos - working to see how he's incorporating the multiple renders he created.

**Reply by tsx-5** [reply]
BTW: I don't have "fill" being used in the CSS file (as you suggested).  But it, in itself, hasn't changed the look at all.

**Reply by OzGav** [reply]
See if there are render engine options still available. I used yarafray when I did mine which had quality options.

**Reply by tsx-5** [reply]
Just a quick update...  

I'm following the video that Trzinka pointed out, which has lead me to reconstruct the Inkscape SVG (2x).  Lesson learned, even though he says you can embed everything in a single SVG (he says you can, but he actually links in video) that creates a very large SVG and HA won't upload files over a certain size (I was at about 18Meg).  To fix this I went to linking the images, but now am getting a blank screen when the SVG is loaded - I'm guessing a lay order issue.  Working on it.

**Reply by OzGav** [reply]
Mine is about 6M last time I looked. Try and scale down before you import so the images aren’t huge to begin with.

**Reply by Trzinka** [reply]
## Hi guys!

`I have image dimensions 2024 x 1400 size 3.85MB and it is on the limit of quality and upload speed!`

**Reply by Trzinka** [reply]
**tsx-5** 
## I wrote to you that the video is for advanced users mainly because it's not "how do I write a clean video (jumps between new and old, uses an old version of Inkscap, etc.)".

`It is best to extract the CSS file first or add commands gradually.`

## In Inkscape, add to the images through the XML Editor add as shown in the image (HA link local path to the files) for each image separately.

![FP-href](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/827c859e-0da6-4f63-876b-435c85363d60)

`It should look like this in devTools:`

![FP-href-DevTools](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/4cbd1af4-1567-4060-a15f-7e057eaf8674)

**Reply by Trzinka** [reply]
**Hi again!**

## Here is a link to my files and settings:
https://github.com/ExperienceLovelace/ha-floorplan/discussions/411

**Reply by tsx-5** [reply]
Excellent job Trzinka - hopefully I can be as useful to you one day, as you are useful to me today.  One question for you...  How do you have your overlay images actually show up in the floor plan?  I have areas setup; can hover over them and see info, click to turn on and off lights, and have the outline of the area change colors based on the on/off state.  but the actual images tied to an overlay aren't showing up - and I don't see any errors when I "inspect" the webpage.  I looked over your code and can't see what drives the images to be loaded.  Could you point that section out?  Thanks again,

**Reply by Trzinka** [reply]
## Hi!

### Sorry for being late, but I am preparing an update https://github.com/ExperienceLovelace/ha-floorplan/discussions/411 where I will (at least I hope) show and explain the process in more detail.

`We are here to share knowledge and help as much as we can!`

**If I understood you correctly, the pictures of the rooms are not displayed when the light is on!**

`I will now update the process on the link mentioned above!`

# Check back in a while or on the go but it will be updated!!

**Reply by Trzinka** [reply]
# It is done!

`I hope I explained in enough detail!`

**Comment by Trzinka** [comment]
https://www.youtube.com/playlist?list=PLNaeA0iHcKdQIfb0TjdS93yqvXJHjTb45

Maybe even the clock when you render: 
`I used the clock a little after sunrise and then lit it. With this, I avoided the shadows created during rendering (unnecessary for the final "even disturbing" look).`

**Reply by Trzinka** [reply]
I would remove the yard with the pillars from the picture, because in all probability you are not controlling anything there.

External lights can be solved differently.

You can also install "Google Dark Theme" in HACS under Frontend (my choice for the floor plan card), if you are a little more skilled, download the "Your Name" (my default theme) theme and adjust it to your favorite colors or "Google Theme - Based on the Android light and dark interface"!

**Reply by tsx-5** [reply]
I definitely look forward to your suggestion on how to address external lights... As I do have a camera and 2 sets of lights I would like to control in this area.

**Reply by Trzinka** [reply]
## ;=))) I haven't gotten to the cameras yet, because there are several solutions (I wish it would work).

### I will look for a solution myself in the sense that it shows me the video only if the camera detects movement (on the left and right side of the screen)!

If you have time, take a look at this page (in Slovenian) where there are examples of different effects. (there's a lot of it) https://www.trzinka.si/Trzinka/
`Naprej = forward`

**Comment by Trzinka** [comment]
# Visit this link!

https://github.com/ExperienceLovelace/ha-floorplan/discussions/411






![Klikni ta dva gumba](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/ad48ecb4-1bb6-4ea4-8f7d-de0e5f35db87)

**Reply by tsx-5** [reply]
Definitely have been visiting your link - lots of great help.  Still working to get my setup straight (work getting in the way of working on this).  Do have a question you don't note - I'm assuming you made your floorplan in SweetHome 3d.  Did you have to crop out the extra area around the house in another app (say like GIMP), before bringing it into Inkscape?

**Reply by OzGav** [reply]
That would be smartest as you can reduce the size of it slightly. I also used Gimp to crop out the light in images and then played with the opacity to get the blending right (I have light spilling between rooms). From there I knew how to layer it in Inkscape and what opacity level to set (if it was different to the standard I was using)

**Reply by tsx-5** [reply]
Thanks OzGav,

With GIMP I note you can crop all your images at once (import them all as layers) but it's not clear to me how to save them back to individual PNG's.  Did you use that function yourself?

**Reply by tsx-5** [reply]
Played with it a bit - and figured it out :-)

**Comment by Trzinka** [comment]
`Definitely have been visiting your link - lots of great help. Still working to get my setup straight (work getting in the way of working on this). Do have a question you don't note - I'm assuming you made your floorplan in SweetHome 3d. Did you have to crop out the extra area around the house in another app (say like GIMP), before bringing it into Inkscape?`


## Yes, unlike you, I really have enough time.
## Yes, I made the floor plan in SweetHome 3d. (I'm not mentioning this because there are really many videos on this topic on YouTube)
## Yes, I first added transparency and then cut out the area around the house in GIMP, and since SweetHome 3d adds edges around the house when rendering, I also cropped the entire image to the size of the house, because I prefer to edit it via CSS.
## Yes, I then transferred the images to Inkscape.

## Maybe take a look on this Q&A page with the filter author: Trzinka in the search field, the other problems I encountered, because I don't know if I turned everything on in https://github.com/ExperienceLovelace/ha-floorplan/discussions/411
